providers/ofrep/src/main/java/dev/openfeature/contrib/providers/ofrep/internal/OfrepApi.java

```diff
@@ -0,0 +1,131 @@
+                .build();
+        this.requestTimeout = requestTimeout;
+        serializer = new ObjectMapper();
+        deserializer = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    }
+
+    /**
+     * prepareHttpRequest is preparing the request to be sent to the OFREP Server.
+     *
+     * @param url         - url of the request
+     * @param headers     - headers to be included in the request
+     * @param requestBody - body of the request
+     *
+     * @return HttpRequest ready to be sent
+     * @throws JsonProcessingException - if an error occurred while processing the json.
+     */
+    private <T> HttpRequest prepareHttpRequest(
+            final URI url, ImmutableMap<String, ImmutableList<String>> headers, final T requestBody)
+            throws JsonProcessingException {
+
+        HttpRequest.Builder reqBuilder = HttpRequest.newBuilder()
+                .uri(url)
+                .timeout(this.requestTimeout)
+                .header("Content-Type", "application/json; charset=utf-8")
+                .POST(HttpRequest.BodyPublishers.ofByteArray(serializer.writeValueAsBytes(requestBody)));
+
+        for (ImmutableMap.Entry<String, ImmutableList<String>> entry : headers.entrySet()) {
+            String key = entry.getKey();
+            for (String value : entry.getValue()) {
+                reqBuilder.header(key, value);
+            }
+        }
+
+        return reqBuilder.build();
+    }
+
+    /**
+     * resolve is the method that interacts with the OFREP server to evaluate flags.
+     *
+     * @param baseUrl      - The base URL of the OFREP server.
+     * @param headers      - headers to include in the request.
+     * @param key          - The flag key to evaluate.
+     * @param requestBody  - The evaluation context as a map of key-value pairs.
+     *
+     * @return Resolution object, containing the response status, headers, and body.
+     */
+    public Resolution resolve(
+            String baseUrl,
+            ImmutableMap<String, ImmutableList<String>> headers,
+            String key,
+            final OfrepRequest requestBody) {
+        if (nextAllowedRequestTime.isAfter(Instant.now())) {
+            throw new GeneralError("Rate limit exceeded. Please wait before making another request.");
+        }
+
+        try {
+            String fullPath = baseUrl + path + key;
+            URI uri = URI.create(fullPath);
+
+            HttpRequest request = prepareHttpRequest(uri, headers, requestBody);
+
+            HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+            OfrepResponse responseBody = deserializer.readValue(response.body(), OfrepResponse.class);
+
+            return new Resolution(response.statusCode(), response.headers(), responseBody);
+        } catch (JsonProcessingException e) {
+            throw new GeneralError("Error processing JSON: " + e.getMessage());
```

Comment:
```suggestion
            throw new ParseError("Error processing JSON: " + e.getMessage());
```

I think this is a good use-case for a `ParseError`.